src/datalogger/src/lib.rs

```diff
@@ -242,8 +259,11 @@ impl DataLogger {
                 {
                     // process a single measurement
                     // is this called a 'single measurement cycle' ?
-
                     self.measure_sensor_values(board); // measureSensorValues(false);
+
+                    let message = DataLogger::receive_usart_message(board);
+                    self.relay_message(message);
```

Comment:
only call relay message if we actually got one (check that the option is not None)